src/app/components/Input/InputCurrency.tsx

```diff
@@ -1,68 +1,44 @@
 import { Currency } from "@arkecosystem/platform-sdk-intl";
-import React, { useCallback, useEffect, useLayoutEffect, useRef, useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
 
 import { Input } from "./Input";
 
 type Props = {
 	addons?: any;
 	onChange?: (value: any) => void;
-	magnitude?: number;
 	as?: React.ElementType<any>;
 } & Omit<React.InputHTMLAttributes<any>, "onChange" | "defaultValue">;
 
-interface ISelectionRange {
-	start: number | null;
-	end: number | null;
-}
+const sanitizeValue = (x?: string) => Currency.fromString(x || "").display;
 
 export const InputCurrency = React.forwardRef<HTMLInputElement, Props>(
-	({ onChange, value, as: Component, magnitude, children, ...props }: Props, ref: any) => {
-		const convertValue = useCallback((value?: string) => Currency.fromString(value || "", magnitude), [magnitude]);
-
-		const [amount, setAmount] = useState<any>(convertValue(value?.toString()));
-		const [selectionRange, setSelectionRange] = useState<ISelectionRange>({
-			start: null,
-			end: null,
-		});
-
-		ref = useRef<HTMLInputElement>();
+	({ onChange, value, as: Component, children, ...props }: Props, ref: any) => {
+		const [amount, setAmount] = useState<string>(sanitizeValue(value?.toString()));
 
 		useEffect(() => {
-			const evaluateValue = (value: any) => {
-				if (value?.display) {
-					return value;
-				}
-
-				return convertValue(value?.toString());
-			};
+			// when value is changed outside, update amount as well
```

Comment:
@goga-m would be nice to wrap this logic into a hook

Reply:
I ended up not adding a custom hook because this component is the only one that would have used it, but also because logic it handles is very simple and short.